apps/developer-hub/src/app/api/search/route.ts

```diff
@@ -1,5 +1,121 @@
-import { createFromSource } from "fumadocs-core/search/server";
+import type { AdvancedIndex } from "fumadocs-core/search/server";
+import { createSearchAPI } from "fumadocs-core/search/server";
+import { z } from "zod";
 
 import { source } from "../../../lib/source";
 
-export const { GET } = createFromSource(source);
+// Define schemas for type safety
+const hermesSchema = z.array(
+  z.object({
+    id: z.string(),
+    attributes: z.object({ symbol: z.string() }),
+  }),
+);
+
+const lazerSchema = z.array(
+  z.object({
+    symbol: z.string(),
+    name: z.string(),
+    pyth_lazer_id: z.number(),
+    description: z.string(),
+  }),
+);
+
+async function getHermesFeeds(): Promise<AdvancedIndex[]> {
+  try {
+    const endpoints = [
+      "https://hermes.pyth.network",
+      "https://hermes-beta.pyth.network",
+    ];
+    const responses = await Promise.all(
+      endpoints.map((url) =>
+        fetch(`${url}/v2/price_feeds`, {
+          next: { revalidate: 3600 },
+        }).then((res) => res.json() as Promise<unknown>),
+      ),
+    );
+
+    const allFeeds: AdvancedIndex[] = [];
+
+    for (const data of responses) {
+      const parsed = hermesSchema.safeParse(data);
+      if (parsed.success) {
+        for (const feed of parsed.data) {
+          allFeeds.push({
+            title: `${feed.attributes.symbol} (Core)`,
+            description: `Price Feed ID: ${feed.id}`,
+            url: `/price-feeds/core/price-feeds/price-feed-ids?search=${feed.attributes.symbol}`,
+            id: feed.id,
+            tag: "price-feed-core",
+            structuredData: {
+              headings: [],
+              contents: [
+                { heading: "Symbol", content: feed.attributes.symbol },
+                { heading: "ID", content: feed.id },
+              ],
+            },
+          });
+        }
+      }
+    }
+
+    return allFeeds;
+  } catch (error: unknown) {
+    throw new Error("Failed to fetch Hermes feeds", { cause: error });
+  }
+}
+
+async function getLazerFeeds(): Promise<AdvancedIndex[]> {
+  try {
+    const res = await fetch(
+      "https://history.pyth-lazer.dourolabs.app/history/v1/symbols",
+      { next: { revalidate: 3600 } },
+    );
+    const data = (await res.json()) as unknown;
+    const parsed = lazerSchema.safeParse(data);
```

Comment:
```suggestion
    const parsed = lazerSchema.safeParse(await res.json());
```

Little trick when using zod schemas with `fetch` that avoids the awkward need to typecast to `unknown`